[Swift in WebKit] Make it possible to use the private SPI module in Swift files in WebKit.framework

rr-codes opened the pull request:
#### 89bfcf5f1e83f651506ce4a9e722e7a1c469531b
<pre>
[Swift in WebKit] Make it possible to use the private SPI module in Swift files in WebKit.framework
<a href="https://bugs.webkit.org/show_bug.cgi?id=282728">https://bugs.webkit.org/show_bug.cgi?id=282728</a>
<a href="https://rdar.apple.com/139394035">rdar://139394035</a>

Reviewed by Abrar Rahman Protyasha and Elliott Williams.

The naming conflict between `Source/WebKit/WebProcess/Inspector/WebInspector.h` and the public `WebInspector.h` header
caused the generated headermap used by the Swift private module to not use the intended header. This caused compile-time
errors when trying to import the private module.

Fix by renaming the source `WebInspector.h` to be `WebInspectorInternal.h`, thereby avoiding the conflict.

* Source/WebKit/Scripts/webkit/messages.py:
(header_for_receiver_name):
(generate_message_handler):
* Source/WebKit/Sources.txt:
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:
* Source/WebKit/WebProcess/InjectedBundle/API/c/WKBundlePage.cpp:
* Source/WebKit/WebProcess/Inspector/WebInspectorClient.cpp:
* Source/WebKit/WebProcess/Inspector/WebInspectorInternal.cpp: Renamed from Source/WebKit/WebProcess/Inspector/WebInspector.cpp.
(WebKit::WebInspector::create):
(WebKit::WebInspector::WebInspector):
(WebKit::WebInspector::~WebInspector):
(WebKit::WebInspector::page const):
(WebKit::WebInspector::openLocalInspectorFrontend):
(WebKit::WebInspector::setFrontendConnection):
(WebKit::WebInspector::closeFrontendConnection):
(WebKit::WebInspector::bringToFront):
(WebKit::WebInspector::whenFrontendConnectionEstablished):
(WebKit::WebInspector::show):
(WebKit::WebInspector::close):
(WebKit::WebInspector::evaluateScriptForTest):
(WebKit::WebInspector::showConsole):
(WebKit::WebInspector::showResources):
(WebKit::WebInspector::showMainResourceForFrame):
(WebKit::WebInspector::startPageProfiling):
(WebKit::WebInspector::stopPageProfiling):
(WebKit::WebInspector::startElementSelection):
(WebKit::WebInspector::stopElementSelection):
(WebKit::WebInspector::elementSelectionChanged):
(WebKit::WebInspector::timelineRecordingChanged):
(WebKit::WebInspector::setDeveloperPreferenceOverride):
(WebKit::WebInspector::setEmulatedConditions):
(WebKit::WebInspector::canAttachWindow):
(WebKit::WebInspector::updateDockingAvailability):
* Source/WebKit/WebProcess/Inspector/WebInspectorInternal.h: Renamed from Source/WebKit/WebProcess/Inspector/WebInspector.h.
(WebKit::WebInspector::setAttached):
(WebKit::WebInspector::disconnectFromPage):
* Source/WebKit/WebProcess/WebPage/WebPage.cpp:
* Source/WebKit/WebProcess/WebPage/mac/WebPageMac.mm:
* Source/WebKit/WebProcess/cocoa/WebProcessCocoa.mm:

Canonical link: <a href="https://commits.webkit.org/286348@main">https://commits.webkit.org/286348@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/006d047508a9c2364192bd5f8901585819a4ee4e

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/75716 "Passed style check") | [⏳ 🛠 ios ](https://ews-build.webkit.org/#/builders/iOS-17-Build-EWS "Waiting in queue, processing has not started yet") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/28567 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/80204 "Built successfully") | [  ~~🛠 win~~](https://ews-build.webkit.org/#/builders/59/builds/26980 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| | [⏳ 🛠 ios-sim ](https://ews-build.webkit.org/#/builders/iOS-17-Simulator-Build-EWS "Waiting in queue, processing has not started yet") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/3004 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/59387 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/59/builds/26980 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/78783 "Passed tests") | [⏳ 🧪 ios-wk2 ](https://ews-build.webkit.org/#/builders/iOS-17-Simulator-Build-EWS "Waiting in queue, processing has not started yet") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/65042 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/39748 "Passed tests") | 
| [✅ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/6/builds/75223 "Passed tests") | [⏳ 🧪 ios-wk2-wpt ](https://ews-build.webkit.org/#/builders/iOS-17-Simulator-Build-EWS "Waiting in queue, processing has not started yet") | | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/25309 "Built successfully") | 
| | [⏳ 🧪 api-ios ](https://ews-build.webkit.org/#/builders/iOS-17-Simulator-Build-EWS "Waiting in queue, processing has not started yet") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/22857 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/81675 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/3055 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/122/builds/1932 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/67616 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/3206 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/65010 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/66925 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/16676 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/10873 "Passed tests") | [  ~~🧪 mac-intel-wk2~~](https://ews-build.webkit.org/#/builders/119/builds/9011 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/3012 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/5834 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/3037 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/3972 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/3044 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

aprotyas commented on Source/WebKit/Scripts/webkit/messages.py:
```diff
@@ -1359,6 +1359,14 @@ def generate_enabled_by_for_receiver(receiver, messages, ignore_invalid_message_
 
 
+def header_for_receiver_name(name):
+    special_headers = {
```
You probably want to leave a comment here (why)? The mapping is very surprising and we should make it known that this is never the precedent to follow for the happy case.

aprotyas commented on Source/WebKit/Scripts/webkit/messages.py:
```diff
@@ -1359,6 +1359,14 @@ def generate_enabled_by_for_receiver(receiver, messages, ignore_invalid_message_
+        'WebInspector': 'WebInspectorInternal'
+    }
+
+    return special_headers[name] or name
```
This is going to raise a `KeyError` if the key `name` is missing from the dictionary.

aprotyas replied:
Ah, EWS clean builds show you the problem.

emw-apple reviewed (approved):
Bummer that we need to do this, but seems like a very reasonable mitigation

rr-codes committed:
[Swift in WebKit] Make it possible to use the private SPI module in Swift files in WebKit.framework
https://bugs.webkit.org/show_bug.cgi?id=282728
rdar://139394035

Reviewed by Abrar Rahman Protyasha and Elliott Williams.

The naming conflict between `Source/WebKit/WebProcess/Inspector/WebInspector.h` and the public `WebInspector.h` header
caused the generated headermap used by the Swift private module to not use the intended header. This caused compile-time
errors when trying to import the private module.

Fix by renaming the source `WebInspector.h` to be `WebInspectorInternal.h`, thereby avoiding the conflict.

* Source/WebKit/Scripts/webkit/messages.py:
(header_for_receiver_name):
(generate_message_handler):
* Source/WebKit/Sources.txt:
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:
* Source/WebKit/WebProcess/InjectedBundle/API/c/WKBundlePage.cpp:
* Source/WebKit/WebProcess/Inspector/WebInspectorClient.cpp:
* Source/WebKit/WebProcess/Inspector/WebInspectorInternal.cpp: Renamed from Source/WebKit/WebProcess/Inspector/WebInspector.cpp.
(WebKit::WebInspector::create):
(WebKit::WebInspector::WebInspector):
(WebKit::WebInspector::~WebInspector):
(WebKit::WebInspector::page const):
(WebKit::WebInspector::openLocalInspectorFrontend):
(WebKit::WebInspector::setFrontendConnection):
(WebKit::WebInspector::closeFrontendConnection):
(WebKit::WebInspector::bringToFront):
(WebKit::WebInspector::whenFrontendConnectionEstablished):
(WebKit::WebInspector::show):
(WebKit::WebInspector::close):
(WebKit::WebInspector::evaluateScriptForTest):
(WebKit::WebInspector::showConsole):
(WebKit::WebInspector::showResources):
(WebKit::WebInspector::showMainResourceForFrame):
(WebKit::WebInspector::startPageProfiling):
(WebKit::WebInspector::stopPageProfiling):
(WebKit::WebInspector::startElementSelection):
(WebKit::WebInspector::stopElementSelection):
(WebKit::WebInspector::elementSelectionChanged):
(WebKit::WebInspector::timelineRecordingChanged):
(WebKit::WebInspector::setDeveloperPreferenceOverride):
(WebKit::WebInspector::setEmulatedConditions):
(WebKit::WebInspector::canAttachWindow):
(WebKit::WebInspector::updateDockingAvailability):
* Source/WebKit/WebProcess/Inspector/WebInspectorInternal.h: Renamed from Source/WebKit/WebProcess/Inspector/WebInspector.h.
(WebKit::WebInspector::setAttached):
(WebKit::WebInspector::disconnectFromPage):
* Source/WebKit/WebProcess/WebPage/WebPage.cpp:
* Source/WebKit/WebProcess/WebPage/mac/WebPageMac.mm:
* Source/WebKit/WebProcess/cocoa/WebProcessCocoa.mm:

Canonical link: https://commits.webkit.org/286348@main
